packages/test/test-end-to-end-tests/src/test/sharedTreeEndToEndTests.spec.ts

```diff
@@ -0,0 +1,461 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils/internal";
+import {
+	ITree,
+	SchemaFactory,
+	TreeViewConfiguration,
+	type TreeView,
+} from "@fluidframework/tree";
+import {
+	asTreeViewAlpha,
+	SharedTree,
+	type Revertible,
+	type TreeViewAlpha,
+} from "@fluidframework/tree/internal";
+
+describeCompat("SharedTree", "NoCompat", (getTestObjectProvider, apis) => {
```

Comment:
How do these tests relate to the ones in packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts ?